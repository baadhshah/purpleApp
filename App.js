// In App.js in a new project
import 'react-native-gesture-handler';
import * as React from 'react';
import Home from './src/screens/home';
import Detail from './src/screens/detail';
import TabView from './src/screens/tab';
import MyDrawer from './src/screens/myDrawer';
import BottomTab from './src/screens/bottomTab';
import TopTab from './src/screens/topTab';
import BottomTabMaterial from './src/screens/bottomTabMaterial';

import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';

const Stack = createNativeStackNavigator();

function App() {
  return (
    <NavigationContainer>
      <Stack.Navigator initialRouteName="Home">
        <Stack.Screen
          name="Home"
          component={Home}
          options={{title: 'My Home'}}
        />
        <Stack.Screen name="Detail" component={Detail} />
        <Stack.Screen name="TabView" component={TabView} />
        <Stack.Screen name="MyDrawer" component={MyDrawer} />
        <Stack.Screen name="BottomTab" component={BottomTab} />
        <Stack.Screen name="TopTab" component={TopTab} />
        <Stack.Screen name="BottomTabMaterial" component={BottomTabMaterial} />
      </Stack.Navigator>
    </NavigationContainer>
  );
}

export default App;
