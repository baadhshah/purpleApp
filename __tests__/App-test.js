/**
 * @format
 */
import 'react-native-gesture-handler';
import 'react-native';
import React from 'react';
import App from '../App';
import Button from '../src/mocks/Button';
import GroceryShoppingList from '../src/mocks/GrocceryShoppingList';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {fireEvent} from '@testing-library/react-native';

it('renders correctly=====', () => {
  renderer.create(<App />);
});

it('renders correctly across screens', () => {
  const tree = renderer.create(<Button />).toJSON();
  console.log('===tree====', tree);
  expect(tree).toMatchSnapshot();
});

test('given empty GroceryShoppingList, user can add an item to it', () => {
  const {getByPlaceholder, getByText, getAllByText} = renderer(
    <GroceryShoppingList />,
  );

  fireEvent.changeText(getByPlaceholder('Enter grocery item'), 'banana');
  fireEvent.press(getByText('Add the item to list'));

  const bananaElements = getAllByText('banana');
  expect(bananaElements).toHaveLength(1); // expect 'banana' to be on the list
});
