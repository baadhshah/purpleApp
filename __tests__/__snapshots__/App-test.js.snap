// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly across screens 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "green",
      "borderRadius": 25,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
      }
    }
  >
    Register
  </Text>
</View>
`;
